cola/widgets/branch.py

```diff
@@ -0,0 +1,520 @@
+                    else:
+                        result = find_recursive(child, child_name)
+                        if result is not None:
+                            return result
+
+                return result
+
+            return find_recursive(top_level_item, name)
+
+        item = find_child(
+            self.topLevelItem(0),
+            self.current_branch)
+
+        if item is not None:
+            self.tree_helper.expand_from_item(item)
+            item.setIcon(0, icons.star())
+
+            tracked_branch = gitcmds.tracked_branch(self.current_branch)
+
+            if self.current_branch is not None and tracked_branch is not None:
+                status = {'ahead': 0, 'behind': 0}
+                status_str = ""
+                args = ["--oneline"]
+
+                origin = tracked_branch + ".." + self.current_branch
+                log = self.git_helper.log(origin, args)
+                status['ahead'] = len(log[1].splitlines())
+
+                origin = self.current_branch + ".." + tracked_branch
+                log = self.git_helper.log(origin, args)
+                status['behind'] = len(log[1].splitlines())
+
+                if status["ahead"] > 0:
+                    status_str += "\u2191" + str(status["ahead"])
+
+                if status["behind"] > 0:
+                    status_str += "  \u2193" + str(status["behind"])
+
+                if status_str != "":
+                    item.setText(0, item.text(0) + "\t" + status_str)
+
+    def git_action_async(self, action, args, kwarg=None, refresh_tree=False,
+                         update_remotes=False):
+        if kwarg is None:
+            kwarg = {}
+
+        task = AsyncGitActionTask(self, self.git_helper, action, args, kwarg,
+                                  refresh_tree, update_remotes)
+        progress = standard.ProgressDialog(N_('Executing action %s') % action,
+                                           N_('Updating'), self)
+
+        self.runtask.start(task, progress=progress,
+                           finish=self.git_action_completed)
+
+    def git_action_completed(self, task):
+        status, out, err = task.result
+        self.git_helper.show_result(task.action, status, out, err)
+
+        if task.refresh_tree:
+            self.refresh()
+
+        if task.update_remotes:
+            model = main.model()
+            model.update_remotes()
+
+    def fetch_action(self):
+        branch = self.tree_helper.get_full_name(self.selected_item(),
+                                                SEPARATOR_CHAR)
+        remote_branch = gitcmds.tracked_branch(branch)
+
+        if remote_branch is not None:
+            rgx = re.compile(r'^(?P<remote>[^/]+)/(?P<branch>.+)$')
+            match = rgx.match(remote_branch)
+
+            if match:
+                remote = match.group('remote')
+                branch_name = match.group('branch')
+                self.git_action_async('fetch', [remote, branch_name])
```

Comment:
With this command I want to update the selected branch with his ref only if its has a tracked branch